pediatric_rootlets/pediatric_rootlets.sh

```diff
@@ -0,0 +1,85 @@
+# - segmentation of vertebral levels from T2w data (sct_label_vertebrae)
+# - detection of PMJ from T2w data (sct_detect_pmj).
+#
+# Authors: Katerina Krejci, Jan Valosek
+#
+
+# Uncomment for full verbose
+set -x
+
+# Immediately exit if error
+set -e -o pipefail
+
+# Exit if user presses CTRL+C (Linux) or CMD+C (OSX)
+trap "echo Caught Keyboard Interrupt within script. Exiting now.; exit" INT
+
+# Retrieve input params
+SUBJECT=$1    # sub-0001/ses-01
+
+# get starting time:
+start=`date +%s`
+
+# SCRIPT STARTS HERE
+# ==============================================================================
+# Display useful info for the log, such as SCT version, RAM and CPU cores available
+sct_check_dependencies -short
+
+# Go to folder where data will be copied and processed
+cd $PATH_DATA_PROCESSED
+
+# Copy source images
+cp -r $PATH_DATA/${SUBJECT%/*} .    # %/* - delete everything after last slash (/)
+
+# Go to anat folder where all structural data are located
+cd ${SUBJECT}/anat
+
+# Run model-spinal-rootlets_ventral_D106_r20240523 for dorsal and ventral rootlets segmentation
+# NOTE: as the model for both ventral and dorsal rootlets is not part of SCT yet, we run directly the nnUNet model using
+# the wrapper script run_inference_single_subject.py from the model-spinal-rootlets repository
+# https://github.com/ivadomed/model-spinal-rootlets/blob/main/packaging_ventral_rootlets/run_inference_single_subject.py
+# NOTE: we use SCT python because it has nnUNet installed 
+# NOTE: the command below expects that you downloaded the model (https://github.com/ivadomed/model-spinal-rootlets/releases/tag/r20240523) and saved it to:  ~/models/model-spinal-rootlets_ventral_D106_r20240523
+
+file_t2=${SUBJECT}_rec-composed_T2w.nii.gz
+
+$SCT_DIR/python/envs/venv_sct/bin/python ~/code/model-spinal-rootlets/packaging_ventral_rootlets/run_inference_single_subject.py -i ${file_t2} -o ${SUBJECT}_T2w_label-rootlets_dseg.nii.gz -path-model ~/models/model-spinal-rootlets_ventral_D106_r20240523/model-spinal-rootlets_ventral_D106_r20240523 -fold all
+
+# Segmentation of spinal cord from T2w data
+# NOTE: on two testing subjects, the contrast agnostic model performed better than sct_deepseg_sc -- it might be
+# interesting to compare the two methods
+sct_deepseg -task seg_sc_contrast_agnostic -i ${file_t2} -o ${SUBJECT}_rec-composed_T2w_deepseg_ca.nii.gz -qc ${PATH_QC} -qc-subject ${SUBJECT}
+
+# Run sct_qc for quality control of rootlet segmentation. We are running the QC here because we need also the SC seg to
+# crop the image
+sct_qc -i ${file_t2} -s ${SUBJECT}_rec-composed_T2w_deepseg_ca.nii.gz -d ${SUBJECT}_T2w_label-rootlets_dseg.nii.gz -p sct_deepseg_lesion -qc ${PATH_QC} -qc-subject ${SUBJECT} -plane axial
+
+# Run sct_label_vertebrae for vertebral levels estimation
+sct_label_vertebrae -i ${file_t2} -s ${SUBJECT}_rec-composed_T2w_deepseg_ca.nii.gz -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT}
```

Comment:
Okay! It seems that we have SC segs and disc labels available under `derivatives`:

<img width="700" alt="image" src="https://github.com/ivadomed/model-spinal-rootlets/assets/39456460/9fcf3eb8-a68e-4cd7-93d3-216e290c3e23">

## TODO:
1. update the script (line 62) to use the existing disc labels; you can get inspiration from the spine-generic [processing script](https://github.com/spine-generic/spine-generic/blob/73e4aa9f5db653b8ef4eca8259dbefe6b0160bb4/process_data.sh#L57) -- @KaterinaKrejci231054 
2. decide what to do with SC segs -- the SC segs under `derivatives` were probably produced by `sct_deeepseg_sc` and manually corrected (by Nadia?) -- I cannot say this for sure because JSON sidecars are missing. Currently, @KaterinaKrejci231054 uses `sct_deepseg -task seg_sc_contrast_agnostic` to segment SC (based on her initial observation, that `contrast-agnostic` performs slightly better than `sct_deeepseg_sc`). This means that if we get new subjects in future, there would be a discrepancy between the method used for SC seg (`sct_deeepseg_sc` vs `contrast-agnostic`). Possible solution: switch from `contrast-agnostic` to `sct_deeepseg_sc` to be consistent (i.e., we would have `sct_deeepseg_sc` + manual corrections).

Reply:
Ad 2) @KaterinaKrejci231054, could you please generate a QC report (`sct_qc`) for 5 random subjects comparing SC seg under derivatives with SC segmented using the contrast-agnostic model?

Reply:
Ad 2) hmm, thinking about SC segs again. Our goal here is to obtain vert labeling. For this, `sct_deepseg_sc` will be fine. --> @KaterinaKrejci231054, you can implement function similar to [this one](https://github.com/spine-generic/spine-generic/blob/73e4aa9f5db653b8ef4eca8259dbefe6b0160bb4/process_data.sh#L83)